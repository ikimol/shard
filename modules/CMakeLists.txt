# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: modules :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

# include CMake helper macros

include(${PROJECT_SOURCE_DIR}/cmake/macros.cmake)
include(${PROJECT_SOURCE_DIR}/cmake/os.cmake)

add_subdirectory(any)
add_subdirectory(math)
add_subdirectory(meta)
add_subdirectory(optional)
add_subdirectory(random)
add_subdirectory(string)
