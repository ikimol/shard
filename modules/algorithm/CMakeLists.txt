# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: modules :: algorithm :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

set(MODULE_NAME algorithm)

set(MODULE_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/modules/algorithm/include)

shard_add_header_only_library(${MODULE_NAME} ${MODULE_INCLUDE_DIR})
