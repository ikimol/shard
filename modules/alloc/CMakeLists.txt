# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: modules :: alloc :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

set(MODULE_NAME alloc)

set(MODULE_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/modules/alloc/include)

shard_add_header_only_library(${MODULE_NAME} ${MODULE_INCLUDE_DIR})

target_link_libraries(${MODULE_NAME} INTERFACE shard::memory)
