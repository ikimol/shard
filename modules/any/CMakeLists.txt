# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: modules :: any :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

set(MODULE_NAME any)

set(MODULE_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/modules/any/include)

shard_add_header_only_library(${MODULE_NAME} ${MODULE_INCLUDE_DIR})

target_link_libraries(${MODULE_NAME} INTERFACE shard::meta)
