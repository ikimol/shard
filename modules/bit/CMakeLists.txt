# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: modules :: bit :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

set(MODULE_NAME bit)

set(MODULE_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/modules/bit/include)

shard_add_header_only_library(${MODULE_NAME} ${MODULE_INCLUDE_DIR})
