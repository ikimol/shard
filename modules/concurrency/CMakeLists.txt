# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: modules :: concurrency :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

set(MODULE_NAME concurrency)

set(MODULE_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/modules/concurrency/include)

shard_add_header_only_library(${MODULE_NAME} ${MODULE_INCLUDE_DIR})

target_link_libraries(${MODULE_NAME} INTERFACE
                      shard::core
                      shard::meta
                      shard::optional
                      )
