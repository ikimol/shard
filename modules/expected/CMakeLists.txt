# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# shard :: modules :: expected :: CMakeLists.txt
# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

set(MODULE_NAME expected)

set(MODULE_INCLUDE_DIR ${PROJECT_SOURCE_DIR}/modules/expected/include)

shard_add_header_only_library(${MODULE_NAME} ${MODULE_INCLUDE_DIR})

target_link_libraries(${MODULE_NAME} INTERFACE
                      shard::memory
                      shard::meta
                      )
